### PR TITLE
Minor performance improvements to basic ImageNet training

### PR DESCRIPTION
**Pull request by ianstenbit**
- Switches to horizontal-only RandomFlip (performed very slightly better in my experiments on MobileNet and EfficientNet)
- Adds global clipnorm
- Increases max epochs to 1000 (training on large TPU clusters with larger batch size is taking more epochs to converge in some cases, setting 1000 as a conservative upper bound)/

### CODE REVIEW

File: examples/training/classification/imagenet/basic_training.py [hunk 1]
```diff
@@ -97,7 +97,7 @@
 
 CLASSES = 1000
 IMAGE_SIZE = (224, 224)
-EPOCHS = 250
+EPOCHS = 1000
```
[the remark this answers is not in the document]
Comment (ianstenbit) [reply examples/training/classification/imagenet/basic_training.py:100]: This script does use EarlyStopping -- the value here is an upper bound

[the remark this answers is not in the document]
Comment (tanzhenyu) [reply examples/training/classification/imagenet/basic_training.py:100]: Maybe put a comment here saying this is an upper bound

File: examples/training/classification/imagenet/basic_training.py [hunk 2]
```diff
@@ -147,7 +147,7 @@
 
 
 AUGMENT_LAYERS = [
-    keras_cv.layers.RandomFlip(),
+    keras_cv.layers.RandomFlip(mode="horizontal"),
```
Comment (tanzhenyu) [inline examples/training/classification/imagenet/basic_training.py:150]: Instead, can we change the default mode for RandomFlip? vertical flip doesn't seem right to me in the first place

File: examples/training/classification/imagenet/basic_training.py [hunk 3]
```diff
@@ -188,8 +188,9 @@ def augment(img, label):
 """
 
 
-optimizer = optimizers.SGD(learning_rate=INITIAL_LEARNING_RATE, momentum=0.9)
-
+optimizer = optimizers.SGD(
+    learning_rate=INITIAL_LEARNING_RATE, momentum=0.9, global_clipnorm=10
```
Comment (tanzhenyu) [inline examples/training/classification/imagenet/basic_training.py:192]: Does this help you to stabilize the training?

Comment (ianstenbit) [reply examples/training/classification/imagenet/basic_training.py:193]: Yes, for a few models I've seen some wild swings in validation accuracy which were resolved with this clipping.

File: examples/training/classification/imagenet/basic_training.py (hunk 2 above)
Comment (ianstenbit) [reply examples/training/classification/imagenet/basic_training.py:151]: Yes, I'll update the default in a separate PR. I think it's nice to make it explicit here anyway.

File: examples/training/classification/imagenet/basic_training.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianstenbit) [reply examples/training/classification/imagenet/basic_training.py:101]: Done

**Review by tanzhenyu — APPROVED** [review]

File: examples/training/classification/imagenet/basic_training.py (hunk 3 above)
Comment (tanzhenyu) [reply examples/training/classification/imagenet/basic_training.py:193]: Maybe this would be where the warmup lr schedule I mentioned earlier would help

**Comment by DavidLandup0** [comment]
Shouldn't the mixed float policy use `bfloat16` instead of `float16` for TPUs? Maybe add that in the accelerator check conditional?

**Comment by tanzhenyu** [comment]
> Shouldn't the mixed float policy use `bfloat16` instead of `float16` for TPUs? Maybe add that in the accelerator check conditional?

Yeah we should do that

**Comment by DavidLandup0** [comment]
Created #908 with change to `float16`, a typo fix and a warmup scheduler and a flag for turning it on and off
